### ISSUE TITLE
map is slow, probably due to markerclusters

### ISSUE THREAD

**Issue by MartinHinz** [xronos-ch/xronos.rails#131]
Currently, the map builds very slow. I guess it is because of the processing of the points in markerclusters. We might consider switching to another library? Here is a paper comparing different solutions:

https://pdfs.semanticscholar.org/dc9f/3e7f230ddf4f321a81d2b027acefa390812f.pdf

Seems like PruneCluster would be the fastest implementation currently?

**Comment by joeroe** [comment]
As discussed earlier, part of the problem is that the JSON needed for the map is just unavoidably big (https://xronos.ch/data.json?&select[]=sites.id&select[]=sites.name&select[]=sites.lat&select[]=sites.lng, ~3.7 MB). Maybe we can stream it in some way, so the map is populated incrementally?

Either way, we should indicate to the user when the data is loading: #133

**Comment by MartinHinz** [comment]
From my checking, the json itself is not so much the problem as it is the display. I switched to "supercluster" in https://github.com/xronos-ch/xronos.rails/tree/faster_map , but it is currently only a proof of concept. Supercluster relies on GeoJSON, and in that branch I convert the JSON into GeoJSON in the browser using Leaflet libraries. It would certainly be faster, if we could serve GeoJSON directly, but I did not do this before discussing with you, @joeroe.
